keps/2937-resource-transformer/README.md

```diff
@@ -0,0 +1,560 @@
+  memory: 100M
+  cpu: 1
+```
+
+To compare, to achieve the identical effective quotas without `replace` the covered
+resource stanza of each ClusterQueue would need to be:
+```yaml
+...
+  - coveredResources: ["cpu", "memory", "example.com/accelerator-memory", "nvidia.com/mig-1g.5gb", "nvidia.com/mig-2g.10gb", "nvidia.com/mig-4g.20gb", "nvidia.com/mig-7g.40gb"]
+    flavors:
+    - name: default-flavor
+      resources:
+      - name: "cpu"
+        nominalQuota: 8
+      - name: "memory"
+        nominalQuota: 500G
+      - name: example.com/accelerator-memory
+        nominalQuota: 200G
+      - name: nvidia.com/mig-1g.5gb
+        nominalQuota: 40  # 200G/5g
+      - name: nvidia.com/mig-2g.10gb
+        nominalQuota: 20  # 200G/10g
+      - name: nvidia.com/mig-4g.20gb
+        nominalQuota: 10  # 200G/20g
+      - name: nvidia.com/mig-7g.40gb
+        nominalQuota: 5  # 200G/40g
+```
+
+##### Story 1B - Using Retain with MIG
+
+However, some cluster admins may desire to use `retain` for MIG resources to exercise
+finer grained control over the use of MIG partitions.
+To do this, the cluster admin could register the following transformations:
+* (`nvidia.com/mig-1g.5gb`, `example.com/accelerator-memory`, `5Gi`, `retain`)
+* (`nvidia.com/mig-2g.10gb`, `example.com/accelerator-memory`, `10Gi`, `retain`)
+* (`nvidia.com/mig-4g.20gb`, `example.com/accelerator-memory`, `20Gi`, `retain`)
+* (`nvidia.com/mig-7g.40gb`, `example.com/accelerator-memory`, `40Gi`, `retain`)
+
+A Job requesting the resources below:
+```yaml
+  nvidia.com/mig-1g.5gb: 2
+  nvidia.com/mig-2g.10gb: 1
+  memory: 100M
+  cpu: 1
+```
+would yield a Workload with an effective resource request of:
+```yaml
+  example.com/accelerator-memory: 20G
+  nvidia.com/mig-1g.5gb: 2
+  nvidia.com/mig-2g.10gb: 1
+  memory: 100M
+  cpu: 1
+```
+
+Below is an example ResourceGroup that the cluster admin could define that
+prevents the usage of `7g.40gb` paritions and puts a tighter cap on `4g.20gb`
+partitions than would be implied by the `accelerator-memory` quota:
+```yaml
+  - coveredResources: ["cpu", "memory", "example.com/accelerator-memory"]
+    flavors:
+    - name: default-flavor
+      resources:
+      - name: "cpu"
+        nominalQuota: 8
+      - name: "memory"
+        nominalQuota: 500G
+      - name: example.com/accelerator-memory
+        nominalQuota: 200G
+      - name: nvidia.com/mig-1g.5gb
+        nominalQuota: 40  # 200G/5g
+      - name: nvidia.com/mig-2g.10gb
+        nominalQuota: 20  # 200G/10g
+      - name: nvidia.com/mig-4g.20gb
+        nominalQuota: 4  # less than the 10 that would be allowed by the acelerator-memory quota
+      - name: nvidia.com/mig-7g.40gb
+        nominalQuota: 0  # not allowed -- inefficient use of GPU resources
+```
+
+[mig]: https://docs.nvidia.com/datacenter/cloud-native/kubernetes/latest/index.html#the-mixed-strategy
+[instaslice]: https://github.com/openshift/instaslice-operator
+
+#### Story 2
+
```

Comment:
This use case might bring us weird metrics and usage status since even though they use only 5 GPUs, the metrics and usage could say using 10 GPUs. 
In large batch clusters, batch admins sometimes expose the cluster capacity and usage using Grafana or other monitoring tools to batch users. In that case, the batch users might be confusing the usage.

So, we may want to allow admins to modify the configurable metrics and usage calculation in the future.
But, in this PR, we can take the feature separately and postpone it to the future based on user feedback.

Reply:
I expanded/clarified user story 2 (credits).  I think this might address this comment.